### PR TITLE
add: code owners support

### PR DESCRIPTION
**Pull request by chdsbd**
With the `reviewDecision` field on the pull request we can fully support code owners. We can rely on this field to determine if a review is needed, which should always cover code ownership.

[`PullRequest` object](https://docs.github.com/en/free-pro-team@latest/graphql/reference/objects#pullrequest), [`PullRequestReviewDecision` enum](https://docs.github.com/en/free-pro-team@latest/graphql/reference/enums#pullrequestreviewdecision)

related #96, #87

### CODE REVIEW

**Comment by chdsbd** [comment]
It seems GitHub's new GraphQL field is broken. I'm getting the following errors:

```
{'errors': [{'message': 'Something went wrong while executing your query. Please include `EECC:1950:28C850:4568A8:5F6E85D2` when reporting this issue.'}]}
```

File: bot/kodiak/evaluation.py [hunk 1]
```diff
@@ -780,6 +781,10 @@ async def set_status(msg: str, markdown_content: Optional[str] = None) -> None:
                 )
                 return
 
+        if pull_request.reviewDecision == PullRequestReviewDecision.REVIEW_REQUIRED:
```
Comment (sindrig) [inline bot/kodiak/evaluation.py:784]: Should this also check for `CHANGES_REQUESTED`? Or maybe `if pull_request.reviewDecision and pull_request.reviewDecision != PullRequestReviewDecision.APPROVED`?

Comment (chdsbd) [reply bot/kodiak/evaluation.py:784]: I think we're covered by the logic that's just above this new code: https://github.com/chdsbd/kodiak/blob/20792c4ccbd22cfe42120203a68d0e1f230b2855/bot/kodiak/evaluation.py#L755-L784

Although, I think we may be able to replace all of that with pull_request.reviewDecision, in which case well need to handle all of the PullRequestReviewDecision states.

**Comment by chdsbd** [comment]
For posterity, my GitHub Support issue for the reviewDecision field bug was escalated to engineering team.

**Comment by chdsbd** [comment]
I sadly haven't heard from GitHub support in the past week.

**Comment by chdsbd** [comment]
GitHub has fixed the bug. I was able to test this change locally and verify that code owners now works.

**Review by sbdchd — APPROVED** [review]

**Comment by chdsbd** [comment]
I'll comment on this PR when this change is deployed this weekend.

**Comment by chdsbd** [comment]
This change has been released with https://github.com/chdsbd/kodiak/releases/tag/v0.30.0.

**Pull request merged**